pkg/syncer/server.go

```diff
@@ -282,7 +282,10 @@ func (s *RegionSyncer) syncHistoryRegion(ctx context.Context, request *pdpb.Sync
 					RegionLeaders: leaders,
 					Buckets:       buckets,
 				}
-				s.limit.WaitN(ctx, resp.Size())
+				if err := s.limit.WaitN(ctx, resp.Size()); err != nil {
+					log.Error("failed to wait rate limit", errs.ZapError(err))
+					return err
```

Comment:
Do we need to return?

Reply:
I think it's possible to return an error here.

After checking the source code, there are two types of errors that WaitN returns, `fmt.Errorf("rate: Wait(n=%d) exceeds limiter's burst %d", n, burst)` and context errors.

- For the former. If `resp.Size()` exceeds the burst, then continuing to loop will still result in an error because `resp.Size` will keep increasing. In this case, the error won't be thrown, but adding logs will keep printing out, and the client may never receive `resp`. Unless we can guarantee that within `maxSyncRegionBatchSize`, resp.Size will never exceed the burst (20MiB).
- For the latter, it doesn't matter, because there is a `select ctx.Done()` inside the loop.

If we can't guarantee that the information of maxSyncRegionBatchSize regions is definitely less than 20MiB, it's best to check if resp.Size is close to 20MiB before continuing at L275.